### ISSUE TITLE
Add regex validation on `Resource.id`

### ISSUE THREAD

**Issue by marcovisserFurore** [FirelyTeam/firely-net-sdk#2163]
**Is your feature request related to a problem? Please describe.**
The type of `Resource.id` is `http://hl7.org/fhirpath/System.String`, which does not have any constraints. But we should apply the constraints of datatype id.

**Describe the solution you'd like**
In the new validator we can add a new validation block for this level.

**Comment by wardweistra** [comment]
Thought reading this: Is this something you prefer to solve specifically for the Resource.id case, or generally for where the `fhir-type` extensions specifies something differing from the actual data type?
(No idea if there are other cases or whether it is generalizable what constraints should be validated from the fhir type)

**Comment by marcovisserFurore** [comment]
Indeed, not specifically for the `Resource.id` case, but also for the other cases, like the fhir-type `uri`.

**Comment by ewoutkramer** [comment]
These properties are exceptional cases in the FHIR serialization. There are only two of them:
* Element.id (a `System.String` that needs to be checked like a FHIR `id`, )
* Extension.uri (a `System.String` that needs to be checked like a FHIR `uri`)

(Primitive.value is different again and actually has specific handling in the parser and the validator)

So we need to do this on *two* spots, with no chance of this ever becoming more.  I think making a general solution for these two cases is much more complex.

**Comment by ewoutkramer** [comment]
I considered `Resource.id` to be inherited from `Element.id`, but I see technically it is not.

**Comment by brianpos** [comment]
Yes, the restrictions on Resource.id (max len 64) are far more restrictive than on Element.id (much longer - used in ElementDefinitions with really long strings)
